1_preprocessing_data/PyBaSiC_Pipelines/scripts/nbconverted/Illumination_Correction.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Illumination Correction of NF1 Images
+
+# ## Import libraries
+
+# In[1]:
+
+
+import sys
+import numpy as np
+import matplotlib
+import pathlib
+from matplotlib import pyplot as plt
+from pathlib import Path
+import os
+import skimage
+
+# explicit import to PyBaSiC due to not having package support
+sys.path.append("./PyBaSiC/")
+import pybasic
+
+
+# ## Create path and load in images
+# The directory set up used for this experiment goes as such:
+# ```
+# 
+# ├── NF1_SchwannCell_data/
+# │   ├── 1_preprocessing_data/
+# │   │   ├── PyBaSiC_Pipelines/
+# │   │   │   ├── NF1_Channels/
+# │   │   │   │   ├── DAPI/ or GFP/ or RFP/
+# 
+# ```
+
+# In[2]:
+
+
+channel_path = '../PyBaSiC_Pipelines/NF1_Channels/RFP/'
+
+# The for loop is running through the files within the folder designated by channel_path and will only take the names of the images and strips the ".tif"
+# which leaves the image name that helps with identification of cells (genotype)
+
+all_files = os.listdir(channel_path)
+file_ext = ".tif"
+image_files = []
+for file in all_files:
+    if file.endswith(file_ext):
+        image_files.append(file.strip(file_ext))
+image_files.sort()
+channel_images = pybasic.tools.load_data(channel_path, '.tif', verbosity = True)
+
+
+# ## Run PyBaSiC to calculate the flatfield and darkfield
+
+# In[3]:
+
+
+flatfield, darkfield = pybasic.basic(channel_images, darkfield=True)
```

Comment:
everything here and below is perfect. It really nicely articulates the process and rationale of each step. Let's keep it in!

However, it only describes one channel, and we should add, before this, the full pipeline to process all channels.

How do you feel about writing a function (similar to as I did in the previous comment) to perform:

1. Flatfield, darkfield calculation
2. Optionally output the flatfield/darkfield images
3. perform illumination correction
4. perform 8 bit conversion
5. save output files

I'd be happy to talk more about this!